### ISSUE TITLE
CI builds does not work: No module named 'zlib'

### ISSUE THREAD

**Issue by tiran** [ethanhs/python-wasm#26]
The CI build artifact does not work:

```
 Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
 Python runtime state: core initialized
 Traceback (most recent call last):
   File "<frozen zipimport>", line 570, in _get_decompress_func
 ModuleNotFoundError: No module named 'zlib'
 
 During handling of the above exception, another exception occurred:

 Traceback (most recent call last):
   File "<frozen zipimport>", line 618, in _get_data
   File "<frozen zipimport>", line 573, in _get_decompress_func
 zipimport.ZipImportError: can't decompress data; zlib not available

 During handling of the above exception, another exception occurred:

 Traceback (most recent call last):
   File "<frozen zipimport>", line 196, in get_code
   File "<frozen zipimport>", line 752, in _get_module_code
   File "<frozen zipimport>", line 620, in _get_data
 zipimport.ZipImportError: can't decompress data; zlib not available
```

There seems to be a problem with the ``Modules/Setup.local`` symlink.

**Comment by emmatyping** [comment]
Looking at the build log, its the same issue I have, which is that for some reason the PYTHONPATH is mixed up between host/build (I think) so imports from io are broken, which breaks extension module building.

```
building '_contextvars' extension
creating build/temp.emscripten-wasm32-3.11/build/cpython/Modules
emcc -Wsign-compare -g -O0 -Wall -std=c99 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I../../Include/internal -I../../Include -IObjects -IPython -I. -I/build/cpython/Include -I/build/cpython/builddir/host -c /build/cpython/Modules/_contextvarsmodule.c -o build/temp.emscripten-wasm32-3.11/build/cpython/Modules/_contextvarsmodule.o
creating build/temp.emscripten-wasm32-3.11/build/cpython/Modules
emcc -Wsign-compare -g -O0 -Wall -std=c99 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I../../Include/internal -I../../Include -IObjects -IPython -I. -I/build/cpython/Include -I/build/cpython/builddir/host -c /build/cpython/Modules/arraymodule.c -o build/temp.emscripten-wasm32-3.11/build/cpython/Modules/arraymodule.o
Fatal Python error: init_sys_streams: can't initialize sys standard streams
Python runtime state: core initialized
Traceback (most recent call last):
  File "/build/cpython/builddir/host/../../Lib/io.py", line 54, in <module>
ImportError: cannot import name 'text_encoding' from 'io' (unknown location)
```

**Comment by emmatyping** [comment]
I think this may have also been the source of my issues in https://bugs.python.org/issue45940

**Comment by tiran** [comment]
That's a red herring. ``setup.py`` is irrelevant for us. Everything should be build by ``make`` rules and linked into the libpython archive file. I think there was an issue with symlink not triggering a rebuild of ``config.c``.

**Comment by tiran** [comment]
I said "*was* an issue" because I think I figured it out. [`488f4e5` (#24)](https://github.com/ethanhs/python-wasm/pull/24/commits/488f4e59d6d3826ba96c6c54b29d01aac24f1f3f)

**Comment by emmatyping** [comment]
Yeah, the setup.py was definitely a red herring. I thought it might be something with the above error I posted (not the error I got in the b.p.o), but it looks like even with that error, things are building correctly! Though we probably should figure out why that error happens, it seems to still happen with the regenerated config.c

```
building '_ctypes' extension
creating build/temp.emscripten-wasm32-3.11/build
creating build/temp.emscripten-wasm32-3.11/build/cpython
creating build/temp.emscripten-wasm32-3.11/build/cpython/Modules
creating build/temp.emscripten-wasm32-3.11/build/cpython/Modules/_ctypes
emcc -Wsign-compare -g -O0 -Wall -std=c99 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I../../Include/internal -I../../Include -IObjects -IPython -I. -I/build/cpython/Include -I/build/cpython/builddir/host -c /build/cpython/Modules/_ctypes/_ctypes.c -o build/temp.emscripten-wasm32-3.11/build/cpython/Modules/_ctypes/_ctypes.o
Fatal Python error: init_sys_streams: can't initialize sys standard streams
Python runtime state: core initialized
Traceback (most recent call last):
  File "/build/cpython/builddir/host/../../Lib/io.py", line 54, in <module>
ImportError: cannot import name 'text_encoding' from 'io' (unknown location)
```

> I said "was an issue" because I think I figured it out. 488f4e5 (#24)

Great find! I wonder why this wasn't an issue before...

**Comment by tiran** [comment]
> Great find! I wonder why this wasn't an issue before...

We both never ran the build script from a clean build dir. Any ``configure`` run after the creation of the symlink creates a correct ``config.c``.

**Comment by emmatyping** [comment]
Hm weird, because I definitely tried deleting my `cpython` folder and building from scratch.

**Comment by emmatyping** [comment]
Anyway, it seems this was fixed in #24
